pkg/provider/aws/resources/lambda_test.go

```diff
@@ -34,44 +30,26 @@ func Test_LambdaCreate(t *testing.T) {
 					{Source: "aws:lambda_function:my-app-test", Destination: "aws:log_group:my-app-test"},
 				},
 			},
+			Check: func(assert *assert.Assertions, lambda *LambdaFunction) {
+				assert.Equal(lambda.Name, "my-app-test")
+				assert.ElementsMatch(lambda.ConstructsRef, []core.AnnotationKey{eu.AnnotationKey})
+			},
 		},
 		{
-			name:    "existing lambda",
-			lambda:  &LambdaFunction{Name: "my-app-test"},
-			wantErr: true,
+			Name:     "existing lambda",
+			Existing: &LambdaFunction{Name: "my-app-test"},
+			WantErr:  true,
 		},
 	}
 	for _, tt := range cases {
-		t.Run(tt.name, func(t *testing.T) {
-
-			assert := assert.New(t)
-			dag := core.NewResourceGraph()
-
-			if tt.lambda != nil {
-				dag.AddResource(tt.lambda)
-			}
-
-			metadata := LambdaCreateParams{
+		t.Run(tt.Name, func(t *testing.T) {
+			tt.Params = LambdaCreateParams{
 				AppName: "my-app",
 				Refs:    []core.AnnotationKey{eu.AnnotationKey},
 				Name:    eu.ID,
 			}
-			lambda := &LambdaFunction{}
-			err := lambda.Create(dag, metadata)
-			if tt.wantErr {
-				assert.Error(err)
-				return
-			}
-			if !assert.NoError(err) {
-				return
-			}
-			tt.want.Assert(t, dag)
-
-			graphLambda := dag.GetResource(lambda.Id())
-			lambda = graphLambda.(*LambdaFunction)
 
-			assert.Equal(lambda.Name, "my-app-test")
-			assert.ElementsMatch(lambda.ConstructsRef, []core.AnnotationKey{eu.AnnotationKey})
+			tt.Run(t)
```

Comment:
can we do other checks on the result in this format or no

Reply:
oh i see now, nevermind, i think this looks good

Reply:
Yeah, and I purposefully made the `Checks func` required (if `!WantError`), so that people don't accidentally forget. I figure even if they want a no-op check, that forces them to at least opt into it.